tools/docker-go

```diff
@@ -54,10 +54,18 @@ DOCKER_RUN_ARGS="--network=host"
 
 parse_args "${@}"
 
+proxy_env=( )
+for i in http_proxy https_proxy no_proxy; do
+   if [ -n "${!i}" ]; then
+      proxy_env[${#proxy_env[@]}]="--env=$i=${!i}"
+   fi
```

Comment:
Do we also need to handle uppercase versions of the proxy environment variables, or are those rarely used?

```suggestion
   if [ -n "${!i}" ]; then
      proxy_env[${#proxy_env[@]}]="--env=$i=${!i}"
   fi
   I="${i^^}"
   if [ -n "${!I}" ]; then
      proxy_env[${#proxy_env[@]}]="--env=${I}=${I}"
   fi
```

Reply:
wow. I'd never seen uppercase translation like that.

In my experience you do not need upper case.  The go stack is happy to read the lower case (and my build succeeded to prove it).  It feels more correct to me to just put one set of variables in, and to know the build system does the right thing with one or the other.

If you'd like to pass them through, I can certainly adjust it to do that.

I think its less code and more readable to just expand the list, though:

    - for i in http_proxy https_proxy no_proxy; do
    + for i in http_proxy https_proxy no_proxy HTTP_PROXY HTTPS_PROXY NO_PROXY; do

Let me know what you prefer and I'll make a change.

Reply:
I'm also happy with the expanded list, if we add a comment like:

```
# Go accepts both uppercase and lowercase variables, so check for both.
```

The only benefit of the uppercase translation is that it enforces this mechanically, but it's also overkill for three variables where we don't expect more to appear.

I think pass-through is valuable for people who have set "HTTP_PROXY" instead of "http_proxy" and expect that to work, even if the lowercase form is more widely known.